packages/untp-test-suite/src/core/processTestSuite.ts

```diff
@@ -14,28 +15,20 @@ import {
   constructFinalReport,
 } from '../templates/utils.js';
 
-export const processCheckDataBySchema = async (
+/**
+ * Function to validate the provided data against a schema.
+ *
+ * @param {object} schema - The schema to validate against.
+ * @param {object} data - The data to be validated.
+ * @param {IConfigContent} credentialConfig - The original credential configuration.
+ * @returns {IValidatedCredentials} The result of the validation with any errors.
+ */
+export const processCheckDataBySchema = (
+  schema: object,
```

Comment:
Why add the schema params to the function? The schema is loaded inside this function, it shouldn't have in the entry params